pkg/reconciler/subscription/controller.go

```diff
@@ -44,10 +46,15 @@ func NewController(
 	subscriptionInformer := subscription.Get(ctx)
 	channelInformer := channel.Get(ctx)
 
+	featureStore := feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"))
```

Comment:
every controller will need to do this too? Is there a better way? Seems like we should at least make `"feature-config-store"` not a magic string.

Reply:
TBH I have no better idea, except having the feature store to be loaded from sharedmain (eg like we do for logging). But, unless I'm mistaken, I think we can't do that because that would require moving all the experimental CM code to pkg, going back to this problem: https://github.com/knative/eventing/pull/5440#discussion_r647141482

Reply:
Ok looking at this comment https://github.com/knative/eventing/pull/5440#discussion_r646946616, I think I get what you mean.

It would be nice to have a context enricher interface, inside the generated reconcile impl, exactly like the validation webhooks, for example:

```
featureStore := feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"))
featureStore.WatchConfigs(cmw)
contextEnricher := featureStore.ToContext

impl := subscriptionreconciler.NewImpl(ctx, r, controller.WithContextEnricher(contextEnricher))
```

This way, the `Reconciler` struct doesn't need the feature flag store and the whole thing just looks better and more consistent with the webhooks API. I can give it a try and see how to implement it.

Reply:
The solution proposed above can be applied with this PR: 

Otherwise, the alternative is to reuse the configStore option (although I'm not sure it's appropriate here?):

```go
featureStore := feature.NewStore(logging.FromContext(ctx).Named("feature-config-store"))
featureStore.WatchConfigs(cmw)

impl := subscriptionreconciler.NewImpl(ctx, r, func(impl *controller.Impl) controller.Options {
    return controller.Options{
	ConfigStore: featureStore,
    }
})
```

Reply:
For the sake of this PR, I've chosen the second alternative, we can reiterate on it after https://github.com/knative/pkg/pull/2147 gets discussed and merged